### PR TITLE
Parse an ellipsis `T...` for type parameter packs

### PR DESCRIPTION
**Pull request by hamishknight**
In a generic parameter list, parse an ellipsis to produce a type parameter pack. This replaces the previous `@_typeSequence` attribute.

### CODE REVIEW

File: include/swift/AST/DiagnosticsParse.def [hunk 1]
```diff
@@ -1964,6 +1964,13 @@ ERROR(attr_requires_concurrency, none,
       "concurrency is enabled",
       (StringRef, bool))
 
+//------------------------------------------------------------------------------
+// MARK: Variadics
+//------------------------------------------------------------------------------
+
+ERROR(associatedtype_cannot_be_pack,none,
+      "associated types may not use type parameter packs", ())
```
Comment (slavapestov) [inline include/swift/AST/DiagnosticsParse.def:1972]: 'associated types cannot be variadic' perhaps?

**Review by slavapestov — APPROVED** [review]
Awesome, thank you!

File: include/swift/AST/DiagnosticsParse.def (hunk 1 above)
Comment (hamishknight) [reply include/swift/AST/DiagnosticsParse.def]: Sounds good to me!

**Comment by hamishknight** [comment]
https://github.com/apple/swift-syntax/pull/948

@swift-ci please test

**Comment by hamishknight** [comment]
One interesting case to consider here is whether to allow `<T ...>`, we actually already allow this for regular variadic parameters e.g `func foo(_ x: Int ...)` in addition to `PackExpansionTypeRepr` e.g `(T ...)`. Should we continue to be consistent with regular variadics here, or should we reject this for pack parameters and pack expansion types?

**Comment by slavapestov** [comment]
I think allowing a space before `...` is fine and would be consistent with the rest of the language

**Comment by hamishknight** [comment]
https://github.com/apple/swift-syntax/pull/948

@swift-ci please test macOS

**Review by CodaFi — APPROVED** [review]

**Pull request merged**